ts-json-schema-generator.ts

```diff
@@ -1,12 +1,14 @@
+import { mkdirSync, writeFileSync } from "node:fs";
+import { dirname } from "node:path";
 import { Command, Option } from "commander";
 import stableStringify from "safe-stable-stringify";
 import { createGenerator } from "./factory/generator.js";
-import { Config } from "./src/Config.js";
+import type { Config } from "./src/Config.js";
 import { BaseError } from "./src/Error/BaseError.js";
 import { formatError } from "./src/Utils/formatError.js";
-import pkg from "./package.json" with { type: "json" };
-import { dirname } from "path";
-import { mkdirSync, writeFileSync } from "fs";
+
+// This constant gets replaced by the build script
+const pkgVersion = "0.0.0"; /* __VERSION__ */
```

Comment:
I don't like this as it makes it harder to watch build. The cjs version used to work so can you try to make the import version work here as well?

Reply:
I tried a lot but ran out of choices.

- `import pkg from 'package.json'` (or * as pkg) works in cjs but not in ESM as it [requires](https://github.com/vega/ts-json-schema-generator/actions/runs/9105923115/job/25032310451#step:7:997) a `with {type: 'json'}` statement

- `require('package.json')` works but not in ESM mode.

- using `const _require = typeof require === 'undefined' ? createRequire(import.meta.url) : require` should work but breaks in CJS even if `createRequire(import.meta.url)` does not gets executed.

- using `fs.readFileSync` requires `__dirname` (which can be created using `import.meta.url` in esm but breaks just like the above try) and a second build step to manually copy package.json to `dist`/`cjs` folders.

- Trying to bypass `import.meta.url` using `eval('import.meta.url') does not work.

I'm not sure if there's any other way to fix it without replacing the dist code. Also, as it gets built on CI before publishing, all official releases should work just as fine.